api/communication/communication-core/src/main/java/jakarta/nosql/ServiceLoaderProvider.java

```diff
@@ -18,23 +18,30 @@
 import java.util.List;
 import java.util.Map;
 import java.util.ServiceLoader;
+import java.util.Spliterator;
 import java.util.WeakHashMap;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.StreamSupport.stream;
 
+import java.lang.reflect.Method;
+
 /**
  * A class that loads class from {@link ServiceLoader}
  */
 public final class ServiceLoaderProvider {
 
     private ServiceLoaderProvider() {
     }
+    
+    private static final String OSGI_SERVICE_LOADER_CLASS_NAME = "org.glassfish.hk2.osgiresourcelocator.ServiceLoader"; //$NON-NLS-1$
 
     private static final Map<Class<?>, Object> CACHE = new WeakHashMap<>();
+    private static Boolean isOsgi = null;
```

Comment:
I believe here makes more sense if you use the AtomicBollean to avoid any race conditions

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicBoolean.html